src/node_process_object.cc

```diff
@@ -216,6 +238,17 @@ void PatchProcessObject(const FunctionCallbackInfo<Value>& args) {
                           env->owns_process_state() ? DebugPortSetter : nullptr,
                           Local<Value>())
             .FromJust());
+
+  // process[exit_code_symbol]
+  CHECK(process
+            ->SetAccessor(context,
```

Comment:
Is this faster than using a typed array?

Reply:
We need to know two things about `process.exitCode`. The first one is its value. The second one is whether it's undefined or not. In the following case, if the `code` isn't an integer type such as `undefined`, then we need to set `kGenericUserError (1)` instead of `kNoFailure (0)`. In other cases, we can use `kNoFailure (0)` for `undefined`.

https://github.com/nodejs/node/blob/3a4f9649304776db7a3a62ebbc0cb477c60c6355/src/node_errors.cc#L1153-L1162

Because of the second one, I couldn't find any other idea except using one more field of a typed array as the following.

```js
  // lib/internal/bootstrap/node.js

  const {
    ...
    exitCode_aliased_Int32Array,
  } = internalBinding('util');

  ...

  const exitCodeAliasedInt32Array = getHiddenValue(process, exitCode_aliased_Int32Array);

  ObjectDefineProperty(process, 'exitCode', {

    ...

    set(code) {
      if (code !== null && code !== undefined) {
        ...
        validateInteger(value, 'code');
        exitCodeAliasedInt32Array[0] = value; // an integer value for exit code
        exitCodeAliasedInt32Array[1] = 1;     // <- To indicate that the exit code is set.
      } else {
        exitCodeAliasedInt32Array[0] = kNoFailure;
        exitCodeAliasedInt32Array[1] = 0;     // <- To indicate that the exit code is unset.
      }
      exitCode = code;
    }
```

I think that using the two fields above may look a bit unclear. So I switched to using `Object::SetAccessor` although it might be a bit slower. Since this part doesn't seem performance critical, I thought this approach could be acceptable. Do you have any other ideas?

Reply:
I thought that the objective, according to the TODO comment linked in the PR description, is to improve performance. Otherwise, what this change does can probably be summarized like this:

* previously, the reading / writing of a JS value was being done from C++ using the V8 API
* now, the reading / writing of a C++ value is being done from JS using object accessors

I feel that the code expresses its intent clearly in both cases, so I guess either way is fine? However, it might be better to know the reason behind switching from the previous approach to the one being proposed here.

Also, if we want to go ahead with this change in its current state, maybe we should leave the typed array TODO comment as is? I don't know if @addaleax has any opinions on this, wdyt?

Reply:
The main objective here is to simplify the code, and a secondary objective to avoid the possibility of reading `process.exitCode` leading to an exception. I would still think that a typed array is the best way to achieve this, but I don’t have strong opinions either way.